src/Compilers/Core/Portable/PEWriter/MetadataWriter.cs

```diff
@@ -3695,6 +3701,24 @@ private void SerializeTypeReference(SignatureTypeEncoder encoder, ITypeReference
                     continue;
                 }
 
+                if (typeReference is IFunctionPointerTypeReference functionPointerTypeReference)
+                {
+                    var signature = functionPointerTypeReference.Signature;
+                    var signatureEncoder = encoder.FunctionPointer(convention: convertCallingConvention(signature.CallingConvention));
+                    SerializeReturnValueAndParameters(signatureEncoder, signature, varargParameters: ImmutableArray<IParameterTypeInformation>.Empty);
+                    return;
+
+                    static SignatureCallingConvention convertCallingConvention(CallingConvention originalConvention)
```

Comment:
>SignatureCallingConvention [](start = 27, length = 26)

It feels like translating between two enums is just another hoop to jump over. Would it be possible to remove the Cci version of the enum and switch to using ```SignatureCallingConvention```, which, I assume, is a public type and could be used in our public APIs as well?